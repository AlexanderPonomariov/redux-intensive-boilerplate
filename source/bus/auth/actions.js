// Types
import { types } from './types';

export const authActions = Object.freeze({
    initialize: () => ({
        type: types.INITIALIZE,
    }),
    authenticate: () => ({
        type: types.AUTHENTICATE,
    }),
    logout: () => ({
        type: types.LOGOUT,
    }),
});
