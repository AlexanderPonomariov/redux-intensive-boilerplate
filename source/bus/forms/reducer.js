// Core
import { combineForms } from 'react-redux-form';

export const formsReducer = combineForms(
    {
        login: {
            email:    'dima1@dima1.com',
            password: '12345',
            remember: true,
        },
        signup: {
            firstName: '',
            lastName:  '',
            email:     '',
            password:  '',
            invite:    'akI2Kynnvwic',
        },
        user: {
            profile: {
                firstName: '',
                lastName:  '',
                avatar:    [],
            },
            password: {
                oldPassword: '',
                newPassword: '',
            },
        },
        feed: {
            comment: '',
        },
    },
    'forms',
);
