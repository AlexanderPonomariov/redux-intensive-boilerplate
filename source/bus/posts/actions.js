// Types
import { types } from './types';

export const postsActions = Object.freeze({
    fillPosts: (posts) => ({
        type:    types.FILL_POSTS,
        payload: posts,
    }),
    createPost: (post) => ({
        type:    types.CREATE_POST,
        payload: post,
    }),
    removePost: (postId) => ({
        type:    types.REMOVE_POST,
        payload: postId,
    }),
    likePost: (likedPostIDs) => ({
        type:    types.LIKE_POST,
        payload: likedPostIDs,
    }),
    unlikePost: (unlikedPostIDs) => ({
        type:    types.UNLIKE_POST,
        payload: unlikedPostIDs,
    }),
    clearPosts: () => ({
        type: types.CLEAR_POSTS,
    }),
});
