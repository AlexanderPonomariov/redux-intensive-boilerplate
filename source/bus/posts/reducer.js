// Instruments
import { types } from './types';

const initialState = [];

export const postsReducer = (state = initialState, action) => {
    switch (action.type) {
        case types.FETCH_POSTS_SUCCESS:

            return action.payload;

        default:
            return state;
    }
};
