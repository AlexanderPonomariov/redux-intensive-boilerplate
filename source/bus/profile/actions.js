// Types
import { types } from './types';

export const profileActions = Object.freeze({
    fillProfile: (user) => ({
        type:    types.FILL_PROFILE,
        payload: user,
    }),
    updateAvatar: (avatarUrl) => ({
        type:    types.UPDATE_AVATAR,
        payload: avatarUrl,
    }),
    clearProfile: () => ({
        type: types.CLEAR_PROFILE,
    }),
});
