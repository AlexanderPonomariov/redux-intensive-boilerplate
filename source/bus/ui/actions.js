// Instruments
import { types } from './types';

export const uiActions = Object.freeze({
    emitError: (error, meta = null) => ({
        type:    types.EMIT_ERROR,
        payload: error,
        error:   true,
        meta,
    }),
    initialize: () => ({
        type: types.INITIALIZE,
    }),
    initializeSuccess: () => ({
        type: types.INITIALIZE_SUCCESS,
    }),
    setPostsFetchingState: (postsFetchingState) => ({
        type:    types.SET_POSTS_FETCHING_STATE,
        payload: postsFetchingState,
    }),
    setAuthFetchingState: (authFetchingState) => ({
        type:    types.SET_AUTH_FETCHING_STATE,
        payload: authFetchingState,
    }),
    setOnlineState: (onlineState) => ({
        type:    types.SET_ONLINE_STATE,
        payload: onlineState,
    }),
    setProfileFetchingState: (state) => ({
        type:    types.SET_PROFILE_FETCHING_STATE,
        payload: state,
    }),
    setProfileEditingState: (state) => ({
        type:    types.SET_PROFILE_EDITING_STATE,
        payload: state,
    }),
    setAvatarFetchingState: (state) => ({
        type:    types.SET_AVATAR_EDITING_STATE,
        payload: state,
    }),
    setPasswordEditingState: (state) => ({
        type:    types.SET_PASSWORD_EDITING_STATE,
        payload: state,
    }),
});
