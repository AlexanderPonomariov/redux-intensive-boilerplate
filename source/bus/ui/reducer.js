// Core
import { Map } from 'immutable';

// Instruments
import { types } from './types';

const initialState = Map({
    isOnline:          false,
    isInitialized:     false,
    isPostsFetching:   false,
    isAuthFetching:    false,
    isProfileFetching: false,
    isProfileEditing:  false,
    isPasswordEditing: false,
    isAvatarFetching:  false,
});

export const uiReducer = (state = initialState, action) => {
    switch (action.type) {
        case types.SET_ONLINE_STATE:
            return state.set('isOnline', action.payload);

        case types.SET_PASSWORD_EDITING_STATE:
            return state.set('isPasswordEditing', action.payload);

        case types.INITIALIZE_SUCCESS:
            return state.set('isInitialized', true);

        case types.SET_POSTS_FETCHING_STATE:
            return state.set('isPostsFetching', action.payload);

        case types.SET_AUTH_FETCHING_STATE:
            return state.set('isAuthFetching', action.payload);

        case types.SET_PROFILE_FETCHING_STATE:
            return state.set('isProfileFetching', action.payload);

        case types.SET_AVATAR_FETCHING_STATE:
            return state.set('isAvatarFetching', action.payload);

        case types.SET_PROFILE_EDITING_STATE:
            return state.set('isProfileEditing', action.payload);

        default:
            return state;
    }
};
