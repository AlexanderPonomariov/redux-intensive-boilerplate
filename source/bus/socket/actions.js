// Instruments
import { socket } from '../../init/socket';
import { uiActions } from '../ui/actions';
import { postsActions } from '../posts/actions';

export const socketActions = Object.freeze({
    listenConnection: () => (dispatch) => {
        socket.on('connect', () => {
            dispatch(uiActions.setOnlineState(true));
        });

        socket.on('disconnect', () => {
            dispatch(uiActions.setOnlineState(false));
        });
    },

    listenPosts: () => (dispatch, getState) => {
        socket.on('create', (response) => {
            const { data, meta } = JSON.parse(response);

            if (getState().profile.get('id') === meta.userID) {
                return;
            }

            dispatch(postsActions.createPost(data));
        });

        socket.on('remove', (response) => {
            const { data, meta } = JSON.parse(response);

            if (getState().profile.get('id') === meta.userID) {
                return;
            }

            dispatch(postsActions.removePost(data));
        });

        socket.on('like', (response) => {
            const { data, meta } = JSON.parse(response);

            if (getState().profile.get('id') === meta.userID) {
                return;
            }

            if (meta.action === 'like') {
                const liker = getState()
                    .users.find((user) => user.get('id') === data.userID)
                    .delete('avatar');

                dispatch(
                    postsActions.likePost({
                        postID: data.postID,
                        liker,
                    })
                );

                return;
            }

            dispatch(postsActions.unlikePost(data));
        });
    },
});
