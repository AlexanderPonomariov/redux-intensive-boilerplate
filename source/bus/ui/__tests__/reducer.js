// Core
import { Map } from 'immutable';

// Instruments
import { uiReducer } from '../reducer';
import { uiActions } from '../actions';

const initialState = Map({
    notification: Map({
        id:      '123',
        type:    'error',
        message: 'Fetch posts fail',
        source:  'fetch posts worker',
    }),
    isOnline:   false,
    isFetching: false,
});

describe('ui reducer:', () => {
    test(`should return initial state by default`, () => {
        expect(uiReducer(void 0, {})).toMatchSnapshot();
    });

    test('should handle setFetchingState action', () => {
        expect(
            uiReducer(void 0, uiActions.setFetchingState(true)),
        ).toMatchSnapshot();
    });

    test('should handle showNotification action', () => {
        expect(
            uiReducer(void 0, uiActions.showNotification('Login success!')),
        ).toMatchSnapshot();
    });

    test('should handle hideNotification action', () => {
        expect(
            uiReducer(initialState, uiActions.hideNotification()),
        ).toMatchSnapshot();
    });

    test('should handle setOnlineState action', () => {
        expect(
            uiReducer(void 0, uiActions.setOnlineState(true)),
        ).toMatchSnapshot();
    });
});
