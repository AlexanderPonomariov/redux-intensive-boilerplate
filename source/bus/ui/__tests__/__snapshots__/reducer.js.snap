// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ui reducer: should handle hideNotification action 1`] = `
Immutable.Map {
  "notification": Immutable.Map {},
  "isOnline": false,
  "isFetching": false,
}
`;

exports[`ui reducer: should handle setFetchingState action 1`] = `
Immutable.Map {
  "notification": Immutable.Map {},
  "isOnline": false,
  "isFetching": true,
}
`;

exports[`ui reducer: should handle setOnlineState action 1`] = `
Immutable.Map {
  "notification": Immutable.Map {},
  "isOnline": true,
  "isFetching": false,
}
`;

exports[`ui reducer: should handle showNotification action 1`] = `
Immutable.Map {
  "notification": Immutable.Map {
    "id": "123",
    "message": "Login success!",
    "type": "info",
    "source": "",
  },
  "isOnline": false,
  "isFetching": false,
}
`;

exports[`ui reducer: should return initial state by default 1`] = `
Immutable.Map {
  "notification": Immutable.Map {},
  "isOnline": false,
  "isFetching": false,
}
`;
